middleware/file/file.go

```diff
@@ -109,14 +110,26 @@ func (f File) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (i
 func (f File) Name() string { return "file" }
 
 // Parse parses the zone in filename and returns a new Zone or an error.
-func Parse(f io.Reader, origin, fileName string) (*Zone, error) {
+// If serial >= 0 it will not reload the zone, it returns an error indicating, nothing
```

Comment:
I think the comment doesn't match the logic.  If serial >= 0 it will not reload the zone, _if the serial is unchanged from last load._